RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc

```diff
@@ -257,11 +257,11 @@ float EcalClusterLazyToolsBase::eseffsirir(const reco::SuperCluster &cluster)
   if (!(fabs(cluster.eta()) > 1.6 && fabs(cluster.eta()) < 3.)) return 0.;
 
   const CaloSubdetectorGeometry *geometryES = geometry_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
-  CaloSubdetectorTopology *topology_p = 0;
-  if (geometryES) topology_p = new EcalPreshowerTopology(geometry_);
+  std::unique_ptr<CaloSubdetectorTopology> topology_p;
+  if (geometryES) topology_p.reset(new EcalPreshowerTopology(geometry_));
 
-  std::vector<float> phoESHitsIXIX = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p, 0, 1);
-  std::vector<float> phoESHitsIYIY = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p, 0, 2);
+  std::vector<float> phoESHitsIXIX = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p.get(), 0, 1);
+  std::vector<float> phoESHitsIYIY = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p.get(), 0, 2);
```

Comment:
Sorry for the delay, I'm at a conference at the moment.

I don't really know in which cases this is called (I just noticed this since it is used in the photon ID module), but if you are right, asserting is certainly wrong.
Your proposal sounds good to me, I'll implement that as soon as I can.